runtime/vm/resolvesupport.cpp

```diff
@@ -496,7 +534,47 @@ resolveStaticMethodRefInto(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA cp
 	if (isResolvedClassAnInterface) {
 		lookupOptions |= J9_LOOK_INTERFACE;
 	}
-	method = (J9Method *)javaLookupMethod(vmStruct, resolvedClass, J9ROMMETHODREF_NAMEANDSIGNATURE(romMethodRef), cpClass, lookupOptions);
+
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	/* Stack allocate a byte array for MethodHandle method name and signature. The array size is:
+	*  - J9ROMNameAndSignature
+	*  - Modified method name
+	*      - U_16 for J9UTF8 length
+	*      - 16 bytes for the original method name ("linkToInterface" is the longest)
+	*  - J9UTF8 for empty signature
+	*/
+	U_8 nameAndNAS[sizeof(J9ROMNameAndSignature) + (sizeof(U_16) + 16) + sizeof(J9UTF8)];
+
+	if (resolvedClass == J9VMJAVALANGINVOKEMETHODHANDLE(vmStruct->javaVM)) {
+		J9UTF8 *nameUTF = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
+		/**
+		 * Check for MH intrinsic methods
+		 *
+		 * Modify the signature to avoid signature mismatch due to varargs
+		 * These methods have special INL send targets
+		 */
+		U_8* initialMethodName = J9UTF8_DATA(nameUTF);
+		U_16 initialMethodNameLength = J9UTF8_LENGTH(nameUTF);
+
+		if (isMethodHandleINL(initialMethodName, initialMethodNameLength)) {
+			J9UTF8 *modifiedMethodName = (J9UTF8 *)(nameAndNAS + sizeof(J9ROMNameAndSignature));
+			J9UTF8 *modifiedMethodSig = (J9UTF8 *)(nameAndNAS + sizeof(nameAndNAS) - sizeof(J9UTF8));
+			memset(nameAndNAS, 0, sizeof(nameAndNAS));
+
+			/* Create new J9ROMNameAndSignature */
+			nameAndSig = (J9ROMNameAndSignature *)nameAndNAS;
+			NNSRP_SET(nameAndSig->name, modifiedMethodName);
+			NNSRP_SET(nameAndSig->signature, modifiedMethodSig);
+
+			modifiedMethodName->length = initialMethodNameLength;
+			memcpy(modifiedMethodName->data, initialMethodName, initialMethodNameLength);
+
+			/* Set flag for partial signature lookup. Signature length is already initialized to 0. */
+			lookupOptions |= J9_LOOK_PARTIAL_SIGNATURE;
```

Comment:
Have you looked at the `J9_LOOK_DIRECT_NAS` lookupOption?  It allows passing direct pointers which would be much simpler:

Then the `onStackNAS` becomes 
```
U_8 onStackNAS[sizeof(J9NameAndSignature)];
J9UTF8 nullSignature = {0};
```
and the assignment becomes:
```
onStackNAS->name = nameAndSig->name;
onStackNAS->signature = nullSignature;
```